pkg/operator/clientutil/clientutil.go

```diff
@@ -107,14 +107,14 @@ func CreateOrUpdateStatefulSet(ctx context.Context, c client.Client, ss *apps_v1
 		ss.SetAnnotations(mergeMaps(ss.Annotations, exist.Annotations))
 
 		err := c.Update(ctx, ss)
-		if k8s_errors.IsNotAcceptable(err) {
+		if k8s_errors.IsNotAcceptable(err) || k8s_errors.IsInvalid(err) {
 			err = c.Delete(ctx, ss)
 			if err != nil {
 				return fmt.Errorf("failed to update statefulset: deleting old statefulset: %w", err)
 			}
 			err = c.Create(ctx, ss)
 			if err != nil {
-				return fmt.Errorf("failed to update statefulset: statefulset: %w", err)
+				return fmt.Errorf("failed to update statefulset: creating new statefulset: %w", err)
```

Comment:
I know this is existing verbiage but these two verbiages sort of feel off. I feel it should be "failed to update statefulset and failed to create new statefulset: %w" or similar.